app/helpers/react_components/training_data/dashboard.rb

```diff
@@ -25,10 +25,14 @@ def training_data_request
       end
 
       def tracks
-        tracks = ::Track.where(id: current_user.user_tracks.trainer.select(:track_id))
-        AssembleTracksForSelect.(tracks)
+        ts = ::Track.where(id: ::TrainingData::CodeTagsSample.where(status: sample_status).select(:track_id))
+        ts = ts.where(id: current_user.user_tracks.trainer.select(:track_id)) unless current_user.staff?
```

Comment:
This does two things:

1. Only show relevant tracks, i.e. the ones for which there is data
2. Only filter the relevant tracks when the user is not staff

CC @dem4ron could you check this works?